frontend/server/src/Controllers/User.php

```diff
@@ -4650,6 +4651,61 @@ public static function apiRevokeAPIToken(\OmegaUp\Request $r) {
         \OmegaUp\DAO\APITokens::deleteByName($r->user->user_id, $name);
         return ['status' => 'ok'];
     }
+
+    /**
+     * @return array{entrypoint: string, templateProperties: array{payload: VerificationParentalTokenDetailsPayload, title: \OmegaUp\TranslationString}}
+     *
+     * @omegaup-request-param string $parental_verification_token
+     */
+    public static function getVerificationParentalTokenDetailsForTypeScript(
+        \OmegaUp\Request $r
+    ): array {
+        $r->ensureIdentity();
+
+        $token = $r->ensureString(
+            'parental_verification_token',
+            validator: fn (string $token) => preg_match(
+                '/^[a-zA-Z0-9]{25}$/',
+                $token
+            ) === 1
+        );
+        $hasParentalVerificationToken = false;
+
+        $userId = \OmegaUp\DAO\Users::findByParentalToken($token);
+
+        if (is_null($userId)) {
+            throw new \OmegaUp\Exceptions\NotFoundException(
+                'parentalTokenNotFound'
+            );
+        }
+
+        if (is_null($r->user) || is_null($r->user->main_email_id)) {
+            throw new \OmegaUp\Exceptions\UnauthorizedException();
+        }
+
+        $user = \OmegaUp\DAO\Users::getByPK($userId);
+        if (is_null($user)) {
+            throw new \OmegaUp\Exceptions\NotFoundException('userNotExist');
+        }
+
+            $user->parent_email_id = $r->user->main_email_id;
+            $user->parent_verified = true;
+            $user->parental_verification_token = null;
+
+            \OmegaUp\DAO\Users::update($user);
+
+            return [
```

Comment:
This could be fixed with the linter, anyways, you could fix it manually
```suggestion
        $user->parent_email_id = $r->user->main_email_id;
        $user->parent_verified = true;
        $user->parental_verification_token = null;

        \OmegaUp\DAO\Users::update($user);

        $hasParentalVerificationToken = true;

        return [
```